pulp_2_tests/tests/rpm/api_v2/test_modular_errata.py

```diff
@@ -0,0 +1,99 @@
+# coding=utf-8
+"""Tests that perform actions over modular Errata repositories."""
+import unittest
+from urllib.parse import urljoin
+from xml.etree import ElementTree
+
+import requests
+from packaging.version import Version
+from pulp_smash import api, config, utils
+from pulp_smash.pulp2.constants import REPOSITORY_PATH
+from pulp_smash.pulp2.utils import (
+    publish_repo,
+    sync_repo,
+)
+
+from pulp_2_tests.constants import (
+    RPM_NAMESPACES,
+    RPM_WITH_MODULES_FEED_URL,
+)
+from pulp_2_tests.tests.rpm.api_v2.utils import (
+    gen_distributor,
+    gen_repo,
+    get_repodata,
+    xml_handler,
+)
+
+
+class ManageModularErrataTestCase(unittest.TestCase):
+    """Manage Modular Eratta content testcase.
+
+    This test targets the following issue.
+
+    `Pulp-2-Tests #94 <https://github.com/PulpQE/Pulp-2-Tests/issues/94>`_
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create class wide variables."""
+        cls.cfg = config.get_config()
+        if cls.cfg.pulp_version < Version('2.17'):
+            raise unittest.SkipTest('This test requires Pulp 2.17 or newer.')
+        cls.client = api.Client(cls.cfg, api.json_handler)
+
+    def test_sync_publish_update_info(self):
+        """Test sync,publish of Modular RPM repo and checks the update info."""
+        body = gen_repo()
+        body['importer_config']['feed'] = RPM_WITH_MODULES_FEED_URL
+        body['distributors'] = [gen_distributor()]
+        repo = self.client.post(REPOSITORY_PATH, body)
+        self.addCleanup(self.client.delete, repo['_href'])
+        sync_repo(self.cfg, repo)
+        repo = self.client.get(repo['_href'], params={'details': True})
+        publish_repo(self.cfg, repo)
```

Comment:
Pulp 2 has the concept of `auto_publish`. We have not explored this one so much.
>auto_publish (boolean) - (optional) if true, this distributor will automatically have its publish operation invoked after a successful repository sync. Defaults to false if unspecified.

[auto_publish](https://docs.pulpproject.org/dev-guide/integration/rest-api/repo/cud.html?highlight=auto_publish%20rpm)

This can be helpful. Try to explore this feature.